docs/core/compatibility/windows-forms/6.0/listview-invalidoperationexception.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Breaking change: ListViewGroupCollection methods throw new InvalidOperationException"
+description: Learn about the breaking change in .NET 6 where some ListViewGroupCollection methods throw a new InvalidOperationException if the ListView is in virtual mode.
+ms.date: 09/23/2021
+---
+# ListViewGroupCollection methods throw new InvalidOperationException
+
+Previously, an <xref:System.InvalidOperationException> was thrown if <xref:System.Windows.Forms.ListViewGroupCollection> methods were invoked on a <xref:System.Windows.Forms.ListViewItem.ListView> in virtual mode *and* the <xref:System.Windows.Forms.Control.Handle> had already been created. Starting in .NET 6, these <xref:System.Windows.Forms.ListViewGroupCollection> methods now only check if the <xref:System.Windows.Forms.ListViewItem.ListView> is in virtual mode. If it is, they throw an <xref:System.InvalidOperationException> with a more descriptive message.
+
+## Previous behavior
+
+Consider the following code that adds a <xref:System.Windows.Forms.ListViewGroup> to a <xref:System.Windows.Forms.ListViewItem.ListView>:
+
+```csharp
+ListViewGroup group1 = new ListViewGroup
+{
+    Header = "CollapsibleGroup1",
+    CollapsedState = ListViewGroupCollapsedState.Expanded
+};
+
+listView.Groups.Add(group1);
+```
+
+This code produced an <xref:System.InvalidOperationException> with the following message:
+
+**When the ListView is in virtual mode, you cannot enumerate through the ListView items collection using an enumerator or call GetEnumerator. Use the ListView items indexer instead and access an item by index value.**
+
+## New behavior
+
+The same code from the [Previous behavior](#previous-behavior) section produces an <xref:System.InvalidOperationException> with the following message:
```

Comment:
The previous/new behavior sections don't show anything about the handle being created or not. Is there a way to show that in the code?

Reply:
@SergeySmirnov-Akvelon to confirm on that.

Reply:
Initially, the issue was reproduced only when the `Handle` was created. The presence of the `Handle` caused an additional thread of code to be called, which tried to access the `Items` and failed because access to `Items` in virtual mode was denied. The user did not see any information about the `Handle`, only an exception with access to Items.

Now, since the virtual mode is always checked, we also display an error only about the virtual mode, without any data about the `Handle`.